internal/impl/packages.go

```diff
@@ -322,16 +326,16 @@ func (d *Devbox) pendingPackagesForInstallation(ctx context.Context) ([]string,
 	if err != nil {
 		return nil, err
 	}
-	for _, pkg := range d.Packages() {
+	for _, input := range d.packagesAsInputs() {
 		_, err := nix.ProfileListIndex(&nix.ProfileListIndexArgs{
 			List:       list,
 			Lockfile:   d.lockfile,
 			Writer:     d.writer,
-			Pkg:        pkg,
+			Input:      input,
 			ProfileDir: profileDir,
 		})
 		if err != nil {
-			pending = append(pending, pkg)
+			pending = append(pending, input.Raw)
```

Comment:
@mikeland86 this if-condition should check `err != nil && errors.Is(err, nix.ErrPackageNotFound)`

Reply:
https://github.com/jetpack-io/devbox/pull/1052